nixos/hardware/hetzner-online/default.nix

```diff
@@ -30,11 +30,9 @@
 
     networking.useNetworkd = true;
     networking.useDHCP = false;
-    # Hetzner servers commonly only have one interface, so its either to just match by that.
-    networking.usePredictableInterfaceNames = false;
 
     systemd.network.networks."10-uplink" = {
-      matchConfig.Name = "eth0";
+      matchConfig.Name = "eno1 ens1 enp1s0 eth0";
```

Comment:
Only question is what we should do if we have multiple interfaces. It's a bit of a special case and maybe we can get away with just documenting it. But if you order a second interface than you end up with two default ipv6 gateways.

I also would than just match for `matchConfig.Type = "ethernet";`.

Reply:
> But if you order a second interface than you end up with two default ipv6 gateways.

I think that's only true if e.g. your first interface is onboard and you second isn't so that you end up with e.g. eno1 & enp1s0. Otherwise you would either end up with eno1 & eno2 or e.g. enp2s0 & enp2s0, were only the former would match.

> I also would than just match for matchConfig.Type = "ethernet";.

But that would match *all* ethernet devices, I deliberately only do a (dirty) match for the first one :)

Reply:
I see.

Reply:
Fine with me than.